engine/gui/src/gui.cpp

```diff
@@ -1477,17 +1499,16 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
             }
         }
 
+        if (num_pruned)
+        {
+            std::sort(c->m_RenderNodes.Begin(), c->m_RenderNodes.End(), RenderEntrySortDisabledPred());
+            c->m_RenderNodes.SetSize(c->m_RenderNodes.Size() - num_pruned);
```

Comment:
Here we make the array the correct size, removing the last entries (which we previously sorted there).